xdsl/dialects/riscv.py

```diff
@@ -211,6 +211,9 @@ def abi_index_by_name(cls) -> dict[str, int]:
     }
 
 
+RD = TypeVar("RD", bound=RISCVRegisterType)
```

Comment:
I'm not sure but it is confusing at first to see why it is `RD`. I would recommend `RdT` for naming conventions for type vars. Better with `RdInvT` as seen in code base.